server/src/main/scala/org/dbpedia/extraction/server/resources/Extraction.scala

```diff
@@ -16,144 +16,443 @@ import org.dbpedia.extraction.destinations.{DeduplicatingDestination, WriterDest
-                  ) yield {
-                    val colon = line.indexOf(':')
-                    (line.substring(2, colon), line.substring(colon + 1, line.length - 2))
-                  }
-                titles.toMap
-            } 
-            finally in.close
-        }
-        catch { 
-            case e : Exception =>
-                logger.log(Level.WARNING, "could not load extraction page titles from classpath resource "+file, e)
-                Map()
+object Extraction {
+  private val logger = Logger.getLogger(getClass.getName)
+
+  val lines: Map[String, String] = {
+    val file = "/extractionPageTitles.txt"
+    try {
+      // ugly - returns null if file not found, which leads to NPE later
+      val in = getClass.getResourceAsStream(file)
+      try {
+        val titles =
+          for (line <- Source.fromInputStream(in)(Codec.UTF8).getLines
+               if line.startsWith("[[") && line.endsWith("]]") && line.contains(':')
+               ) yield {
+            val colon = line.indexOf(':')
+            (line.substring(2, colon), line.substring(colon + 1, line.length - 2))
+          }
+        titles.toMap
+      }
+      finally in.close
+    }
+    catch {
+      case e: Exception =>
+        logger.log(Level.WARNING, "could not load extraction page titles from classpath resource " + file, e)
+        Map()
+    }
+  }
+
+  // Load configuration from properties files
+  private lazy val configProperties: Properties = {
+    val props = new Properties()
+
+    // Try loading server.properties first, then server.default.properties
+    val configFiles = Seq("server.properties", "server.default.properties")
+    var loaded = false
+
+    for (configFile <- configFiles if !loaded) {
+      try {
+        val file = new File(configFile)
+        if (file.exists()) {
+          logger.info(s"Loading configuration from $configFile")
+          val fis = new FileInputStream(file)
+          try {
+            props.load(fis)
+            loaded = true
+            logger.info(s"Successfully loaded configuration from $configFile")
+          } finally {
+            fis.close()
+          }
         }
+      } catch {
+        case e: Exception =>
+          logger.log(Level.WARNING, s"Could not load $configFile", e)
+      }
     }
+
+    if (!loaded) {
+      logger.warning("No configuration file found. Checked: " + configFiles.mkString(", "))
+    }
+
+    props
+  }
+
+  /**
+   * Get available extractors for a specific language from the server configuration
+   */
+  def getAvailableExtractors(language: Language): Seq[String] = {
+    try {
+      // Use Server's method to get available extractors
+      Server.instance.getAvailableExtractorNames(language)
+    } catch {
+      case e: Exception =>
+        logger.log(Level.WARNING, s"Could not get extractors for language ${language.wikiCode}", e)
+        getDefaultExtractorsForLanguage(language.wikiCode)
```

Comment:
we could keep this for debugging and testing the code if ypou find it useful, but we should remove the related code before merging, since the actual use may vary a lot. 
If the language is enabled in the config and we get no extractors back, we should raise an error with a related message
if the language is not enabled in the config, we should raise an error with a related message